test/metabase/lib/filter_test.cljc

```diff
@@ -549,6 +546,27 @@
       {:clause [:not-null created-at],
        :name "Created At is not empty"}])))
 
+(deftest ^:parallel day-of-week-frontend-filter-display-names-test
+  (let [created-at (meta/field-metadata :products :created-at)
+        created-at-with #(lib/with-temporal-bucket created-at %1)]
+    (check-display-names
+     [{:clause [:= (created-at-with :day-of-week) "2023-10-02"],
+       :name "Created At: Day of week is Monday"}
+      {:clause [:= (lib.expression/get-day-of-week created-at :iso) 1],
+       :name "Created At is Monday"}
+      {:clause [:= (created-at-with :day-of-week) "2023-10-02" "2023-10-03" "2023-10-04"],
+       :name "Created At: Day of week is 3 selections"}
+      {:clause [:= (lib.expression/get-day-of-week created-at :iso) 1 2 3],
+       :name "Created At is 3 selections"}
```

Comment:
The new equality filter gets a display name without time bucket.

Reply:
I omitted it, because I think it can suggest bucketing, but we are not doing that. In most cases it should be understandable from the rest of the display name. My last change updates these, please have a look if it makes sense to you.